crypto/threads_pthread.c

```diff
@@ -58,9 +58,13 @@ CRYPTO_RWLOCK *CRYPTO_THREAD_lock_new(void)
      */
     pthread_mutexattr_init(&attr);
 #  if !defined(NDEBUG) && !defined(OPENSSL_NO_MUTEX_ERRORCHECK)
+#   if defined(PTHREAD_MUTEX_ERRORCHECK)
     pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_ERRORCHECK);
-# else
+#   endif
+#  else
+#   if defined(PTHREAD_MUTEX_NORMAL)
     pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_NORMAL);
+#   endif
```

Comment:
I am sorry for confusing things but unfortunately I realized the PTHREAD_MUTEX_NORMAL/PTHREAD_MUTEX_ERRORCHECK are not macros but just enum values. So this will not work. So please just revert this part to your original solution (possibly with inverting the condition and else).

Reply:
> 
> 
> I am sorry for confusing things but unfortunately I realized the PTHREAD_MUTEX_NORMAL/PTHREAD_MUTEX_ERRORCHECK are not macros but just enum values. So this will not work. So please just revert this part to your original solution (possibly with inverting the condition and else).

Will correct. Note that these are macros under cygwin as well. My Ubuntu system has them as enums. The set of values in cygwin - pretty recent update - does not line up precisely with Ubuntu.